.github/workflows/nextjs.yml

```diff
@@ -0,0 +1,95 @@
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: 'pages'
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Detect package manager
+        id: detect-package-manager
+        run: |
+          if [ -f "${{ github.workspace }}/yarn.lock" ]; then
+            echo "manager=yarn" >> $GITHUB_OUTPUT
+            echo "command=install" >> $GITHUB_OUTPUT
+            echo "runner=yarn" >> $GITHUB_OUTPUT
+            exit 0
+          elif [ -f "${{ github.workspace }}/package.json" ]; then
+            echo "manager=npm" >> $GITHUB_OUTPUT
+            echo "command=ci" >> $GITHUB_OUTPUT
+            echo "runner=npx --no-install" >> $GITHUB_OUTPUT
+            exit 0
+          else
+            echo "Unable to determine package manager"
+            exit 1
+          fi
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+          cache: ${{ steps.detect-package-manager.outputs.manager }}
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+        with:
+          # Automatically inject basePath in your Next.js configuration file and disable
+          # server side image optimization (https://nextjs.org/docs/api-reference/next/image#unoptimized).
+          #
+          # You may remove this line if you want to manage the configuration yourself.
+          static_site_generator: next
+      - name: Restore cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            .next/cache
+          # Generate a new cache whenever packages or source files change.
+          key: ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-${{ hashFiles('**.[jt]s', '**.[jt]sx') }}
+          # If source files changed but packages didn't, rebuild from a prior cache.
+          restore-keys: |
+            ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json', '**/yarn.lock') }}-
+      - name: Install dependencies
+        run: ${{ steps.detect-package-manager.outputs.manager }} ${{ steps.detect-package-manager.outputs.command }}
+      - name: Build with Next.js
+        run: ${{ steps.detect-package-manager.outputs.runner }} next build
+      - name: Static HTML export with Next.js
+        run: ${{ steps.detect-package-manager.outputs.runner }} next export
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v2
+        with:
+          path: ./out
+
+  # Deployment job
+  deploy:
```

Comment:
@justin-kleid I think this deploy should only be allowed when merging to main branch, because we don't want to deploy every pull request to Github Pages.  For a pull request, I think we need to deploy to github pages preview: https://github.com/marketplace/actions/deploy-pr-preview

You can use condition to detect if merge a PR: if: github.event.pull_request.merged == true

Reply:
Shouldn't the deploy preview check not be failing?

It throws the error when I manually try to deploy pr preview. Not sure if that's because it is not being done on an actual pull request or not.

error:

Checking configuration and starting deployment… 🚦
Deploying using Deploy Token… 🔑
Configuring git…
/usr/bin/git config --global --add safe.directory /home/runner/work/reactgeoda/reactgeoda
/usr/bin/git config user.name lixun910
fatal: not in a git directory
Error: There was an error initializing the repository: The process '/usr/bin/git' failed with exit code [128](https://github.com/GeoDaCenter/reactgeoda/actions/runs/5730065133/job/15528104943#step:3:133) ❌
Notice: Deployment failed! ❌

Reply:
The current error is now: Seems to be a permission issue, I think the github token needs write access for the line:
"Run marocchino/sticky-pull-request-comment@v2"

remote: Permission to GeoDaCenter/reactgeoda.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/GeoDaCenter/reactgeoda.git/': The requested URL returned error: 403
Changes committed to the gh-pages branch… 📦
Running post deployment cleanup jobs… 🗑️
/usr/bin/git checkout -B github-pages-deploy-action/3737ea2si
Reset branch 'github-pages-deploy-action/3737ea2si'
/usr/bin/chmod -R +rw github-pages-deploy-action-temp-deployment-folder
/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
Completed deployment successfully! ✅
Run echo "url=https://geodacenter.github.io/reactgeoda/pr-preview/pr-24/"  >> $GITHUB_OUTPUT
Run marocchino/sticky-pull-request-comment@v2
Error: Resource not accessible by integration

Reply:
Got it. Let me check it.

Reply:
See https://github.com/marocchino/sticky-pull-request-comment#error-resource-not-accessible-by-integration